mozilla_django_oidc_db/backends.py

```diff
@@ -45,33 +45,41 @@ def get_user_instance_values(self, claims):
 
     def create_user(self, claims):
         """Return object for a newly created user account."""
-        sub = claims.get("sub")
+        username_claim = self.config.username_claim
+        unique_id = claims.get(username_claim)
 
-        logger.debug("Creating OIDC user: %s", sub)
+        logger.debug("Creating OIDC user: %s", unique_id)
 
         user = self.UserModel.objects.create_user(
-            **{self.UserModel.USERNAME_FIELD: sub}
+            **{self.UserModel.USERNAME_FIELD: unique_id}
         )
         self.update_user(user, claims)
 
         return user
 
     def filter_users_by_claims(self, claims):
         """Return all users matching the specified subject."""
-        sub = claims.get("sub")
+        username_claim = self.config.username_claim
+        unique_id = claims.get(username_claim)
 
-        if not sub:
+        if not unique_id:
             return self.UserModel.objects.none()
-        return self.UserModel.objects.filter(username__iexact=sub)
+        return self.UserModel.objects.filter(
+            **{f"{self.UserModel.USERNAME_FIELD}__iexact": unique_id}
+        )
 
     def verify_claims(self, claims):
         """Verify the provided claims to decide if authentication should be allowed."""
         scopes = self.get_settings("OIDC_RP_SCOPES", "openid email")
 
         logger.debug("OIDC claims received: %s", claims)
 
-        if "sub" not in claims:
-            logger.error("`sub` not in OIDC claims, cannot proceed with authentication")
+        username_claim = self.config.username_claim
+
+        if username_claim not in claims:
+            logger.error(
+                f"`{username_claim}` not in OIDC claims, cannot proceed with authentication"
+            )
```

Comment:
```suggestion
            logger.error(
                "%s not in OIDC claims, cannot proceed with authentication", username_claim
            )
```
don't want your app to crash if logging interpolation fails